### PR TITLE
Update telemetry 

### PR DESCRIPTION
**Pull request by manodnyab**
## Description

1) Adding a credentialSourceId to the CredentialProviderFactory
2) Changing region.id to awsRegion using MetricEventMetadata to capture region 

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CredentialSourceId.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import software.aws.toolkits.core.credentials.CredentialSourceId
+import software.aws.toolkits.telemetry.CredentialSourceId as TelemetryCredentialSourceId
+
+fun CredentialSourceId?.toTelemetryCredentialSourceId(): TelemetryCredentialSourceId = when (this) {
+    CredentialSourceId.SharedCredentials -> TelemetryCredentialSourceId.SharedCredentials
+    CredentialSourceId.SdkStore -> TelemetryCredentialSourceId.SdkStore
+    CredentialSourceId.Ec2 -> TelemetryCredentialSourceId.Ec2
+    CredentialSourceId.EnvVars -> TelemetryCredentialSourceId.EnvVars
+    null -> TelemetryCredentialSourceId.Other
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CredentialSourceId.kt:14]: We should make this "else" branch

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateFunctionDialog.kt [hunk 2]
```diff
@@ -70,13 +73,21 @@ class CreateFunctionDialog(private val project: Project, private val initialRunt
...
-            initialDeploy = true
-        )
+        val connectionSettings = AwsConnectionManager.getInstance(project).connectionSettings()
+        val account = connectionSettings?.let { AwsResourceCache.getInstance().getResourceIfPresent(StsResources.ACCOUNT, it) }
+        account?.let { it ->
+            MetricEventMetadata(
+                awsAccount = it,
+                awsRegion = connectionSettings.region.toString()
+            )
+        }?.let { metricEventMetadata ->
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateFunctionDialog.kt:83]: We should always push the telemetry, even if we can't get the info.

See:
```
software.aws.toolkits.jetbrains.services.telemetry.TelemetryService#record
```

File: core/src/software/aws/toolkits/core/credentials/ToolkitCredentialsProvider.kt [hunk 3]
```diff
@@ -14,6 +14,13 @@ enum class CredentialType {
     SsoProfile
 }
 
+enum class CredentialSourceId {
```
Comment (hunterwerlla) [inline core/src/software/aws/toolkits/core/credentials/ToolkitCredentialsProvider.kt:17]: Since this is only for telemetry, do we actually need this?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateFunctionDialog.kt (hunk 2 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateFunctionDialog.kt:83]: Maybe we should expose this logic as a public method on TelemetryService to encapsulate it,
something like

`TelemetryService#calcualteMetadata(project: Project, connectionSettingsOverride: ConnectionSettings? = null) `

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/UploadObjectAction.kt [hunk 4]
```diff
@@ -70,9 +70,9 @@ fun uploadObjects(project: Project, treeTable: S3TreeTable, files: List<VirtualF
                 }
             }
 
-            S3Telemetry.uploadObjects(project, Result.Succeeded, files.size.toDouble())
+            S3Telemetry.uploadObjects(project, Result.Succeeded)
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/UploadObjectAction.kt:73]: Why were these changed?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateFunctionDialog.kt (hunk 2 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateFunctionDialog.kt:83]: Re-reading, why are we making the metadata manually when this is just the project settings?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/DeleteTagAction.kt [hunk 5]
```diff
@@ -78,14 +78,14 @@ class DeleteTagAction : ExplorerNodeAction<EcrTagNode>(message("ecr.delete.tag.a
                             content = message("ecr.delete.tag.succeeded", selected.size, repositoryName)
                         )
                         project.refreshAwsTree(EcrResources.listTags(repositoryName))
-                        EcrTelemetry.deleteTags(project, Result.Succeeded, selected.size.toDouble())
+                        EcrTelemetry.deleteTags(project, Result.Succeeded)
```
Comment (abrooksv) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/DeleteTagAction.kt:81]: Why did this get changed?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateFunctionDialog.kt (hunk 2 above)
Comment (manodnyab) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateFunctionDialog.kt]: Changed it to project, wasn't sure if the region was accounted for earlier

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CredentialSourceId.kt (hunk 1 above)
Comment (manodnyab) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/CredentialSourceId.kt]: done

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/UploadObjectAction.kt (hunk 4 above)
Comment (manodnyab) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/UploadObjectAction.kt:73]: Their definition was changed.

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/UploadObjectAction.kt:73]: File files size should map to value, so you just need to switch to named args

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/DeleteTagAction.kt (hunk 5 above)
Comment (manodnyab) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecr/actions/DeleteTagAction.kt]: updated the metric

**Review by abrooksv — APPROVED** [review]

File: core/src/software/aws/toolkits/core/credentials/ToolkitCredentialsProvider.kt (hunk 3 above)
Comment (abrooksv) [reply core/src/software/aws/toolkits/core/credentials/ToolkitCredentialsProvider.kt:17]: We define credentials factory in Core, but we generate telemetry in JB Core, so we either:
1. go unsafe and use strings
2. Dupe the enum for some semblance of type safety.

I voted for 2.
